src/Workspaces/CSharp/Portable/CodeGeneration/CSharpCodeGenerationHelpers.cs

```diff
@@ -21,6 +23,133 @@ namespace Microsoft.CodeAnalysis.CSharp.CodeGeneration
 {
     internal static class CSharpCodeGenerationHelpers
     {
+        public static (ImmutableArray<AttributeData>, NullableAnnotation) AdjustNullableAnnotationByAttributes(ITypeSymbol type, RefKind refKind, ImmutableArray<AttributeData> attributes, NullableAnnotation nullableAnnotation, bool isParameter)
+        {
+            if (type.IsReferenceType)
+            {
+                // The compiler does not allow 'T?' where 'T : class?'.
```

Comment:
I'm not sure what this means:

![image](https://user-images.githubusercontent.com/4564579/115928848-a698f580-a43b-11eb-8c6e-1b9255a559bb.png)